aten/src/ATen/ParallelThreadPoolNative.cpp

```diff
@@ -67,10 +68,17 @@ int get_num_interop_threads() {
 }
 
 void launch(std::function<void()> func) {
+  auto debug_info = getThreadLocalDebugInfo();
+  auto fn = [func, debug_info]() {
+    auto prev_info = getThreadLocalDebugInfo();
+    setThreadLocalDebugInfo(debug_info);
```

Comment:
nit: std::move?

also you can make this API to return the previous version (do a swap)